task.sql

```diff
@@ -2,0 +1,37 @@
+CREATE DATABASE ShopDB;
+USE ShopDB;
+
+CREATE TABLE Products (
+    ID INT AUTO_INCREMENT,
+    Name VARCHAR(50),
+    Description VARCHAR(100),
+    Price DECIMAL(10, 2),
+    WarehouseAmount INT,
+    PRIMARY KEY (ID)
+);
+
+CREATE TABLE Customers (
+		ID INT AUTO_INCREMENT,
+		FirstName VARCHAR(50),
+		LastName VARCHAR(50),
+		Email VARCHAR(100),
+		Address VARCHAR(100),
+    PRIMARY KEY (ID)
+);
+
+CREATE TABLE Orders (
+		ID INT AUTO_INCREMENT,
+		CustomerID INT NULL,
+		Date DATE,
+		PRIMARY KEY (ID),
+		FOREIGN KEY (CustomerID) REFERENCES Customers(ID) ON DELETE SET NULL
+);
+
+CREATE TABLE OrderItems (
+		ID INT AUTO_INCREMENT,
+		OrderID INT NULL,
+		ProductID INT NULL,
```

Comment:
Consider making the `ProductID` column in the `OrderItems` table NOT NULL if every order item must be associated with a product. This will enforce data integrity by ensuring that each order item is linked to a valid product.